### PR TITLE
Update sailsjs configuration format to support v1.

### PR DESCRIPTION
**Pull request by mrded**
`sails.config.connections` -> `sails.config.datastores`
`sails.config.models.connection` -> `sails.config.models.datastore`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -50,15 +50,16 @@ module.exports = function (sails) {
 
         initConnections: function () {
             var connections = {}, connection, connectionName;
+            var datastore = sails.config.models.datastore || 'default';
```
Comment (KSDaemon) [inline index.js]: Hmm.... Why not to define a key for searching connection properties, like
```
var connectionKey = sails.config.models.datastore ? 'datastore' : 'connection';
```

File: index.js [hunk 2]
```diff
@@ -50,15 +50,16 @@ module.exports = function (sails) {
...
-            sails.log.verbose('Using default connection named ' + sails.config.models.connection);
-            if (!sails.config.connections[sails.config.models.connection]) {
-                throw new Error('Default connection \'' + sails.config.models.connection + '\' not found in config/connections');
+            sails.log.verbose('Using default connection named ' + datastore);
+            if (!sails.config.datastores[datastore]) {
+                throw new Error('Default connection \'' + datastore + '\' not found in config/connections');
             }
 
-            for (connectionName in sails.config.connections) {
+            for (connectionName in sails.config.datastores) {
```
Comment (KSDaemon) [inline index.js]: And then use it everywhere like here:
```
for (connectionName in sails.config[connectionKey + 's']) {
...
```
This allow us for supporting both pre v1.0 versions along with v1.0

File: index.js [hunk 3]
```diff
@@ -96,7 +97,7 @@ module.exports = function (sails) {
                 modelDef = models[modelName];
                 sails.log.verbose('Loading model \'' + modelDef.globalId + '\'');
 
-                connectionName = modelDef.options.connection || sails.config.models.connection;
+                connectionName = modelDef.datastore || sails.config.models.datastore || 'default';
```
Comment (KSDaemon) [inline index.js]: in v1.0 datastore is now located in model definition root? (instead of .options, like in v0.12) ?

**Comment by mrded** [comment]
I tried to run tests on `v.1.1.0` version without my changes, and all tests fail the same way. Are you sure that tests are working?

**Comment by KSDaemon** [comment]
Hmm... Yes, test are working as on mine two computers, also on travis-ci. You can check `Details` link

File: index.js [hunk 4]
```diff
@@ -92,11 +97,15 @@ module.exports = function (sails) {
...
+            // Try to read settings from old Sails then from the new.
+            // 0.12: sails.config.models.connection
+            // 1.00: sails.config.models.datastore
+            var defaultConnection = sails.config.models.connection || sails.config.models.datastore || 'default';
             for (modelName in models) {
                 modelDef = models[modelName];
                 sails.log.verbose('Loading model \'' + modelDef.globalId + '\'');
 
-                connectionName = modelDef.options.connection || sails.config.models.connection;
+                connectionName = modelDef.options.connection || defaultConnection;
```
Comment (KSDaemon) [inline index.js]: Don't you forget about model definition datastore variant for v1.0? :)

File: index.js (hunk 3 above)
Comment (mrded) [reply index.js]: I believe so http://next.sailsjs.com/documentation/concepts/models-and-orm/model-settings#?datastore

File: index.js (hunk 4 above)
Comment (mrded) [reply index.js]: No, it in :103 line.

File: index.js (hunk 1 above)
Comment (mrded) [reply index.js]: I've updated it already to support both variants of sails. Please have a look.

File: index.js (hunk 4 above)
Comment (KSDaemon) [reply index.js]: I mean is `modelDef.options.connection` correct? Or it should be `modelDef.options.datastore` ?

Comment (mrded) [reply index.js]: Oh, I see. You're right. I forgot `modelDef.datastore`

**Comment by mrded** [comment]
> Also please add working tests for your changes. It will be great, if we can create a tests both for v0.12 and v1.0 version.

I'm not really sure how to make tests for different version of packages. Can you help me with that?

**Comment by mrded** [comment]
Great, thank you!

**Pull request merged**